crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -704,6 +705,21 @@ impl<'a> Resolver<'a> {
         }
     }
 
+    fn handle_is_function_internal(&mut self, func: &NoirFunction) -> Option<bool> {
+        if func.def.is_internal {
+            if self.in_contract() {
+                Some(true)
+            } else {
+                self.push_err(ResolverError::ContractFunctionInternalInNormalFunction {
+                    span: func.name_ident().span(),
+                });
+                None
+            }
+        } else {
+            Some(false)
+        }
```

Comment:
I think it is slightly easier to track which cases are Some if the cases are re-ordered:
```suggestion
if self.in_contract() {
    Some(func.def.is_internal)
} else {
    if func.def.is_internal {
        self.push_err(ResolverError::ContractFunctionInternalInNormalFunction {
        span: func.name_ident().span(),
    });
    None
}
```